src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -145,22 +145,43 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     // processor needed for BBCode to HTML translation
     TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("kefirbb.xml");
 
-    String messageContent = request.getParameter("message");
+    String messageContent;
+    if(request.getParameter("message") != null)
```

Comment:
Since the reply form specifies the parent message UUID, I think you could just have both forms use the "message" parameter for the message text. It would eliminate the need to have this conditional logic here.